src/components/CChartSummery/index.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import BN from 'src/utils/BN';
+import classNames from 'classnames';
+
+import getPercentageColor from './getPercentageColor';
+
+import Arrow from '@/assets/Arrow';
+
+interface CChartSummeryProps {
+  text?: string | number;
+  percentage?: number;
+  className?: string;
+  transparent?: boolean;
+}
+
+const CChartSummery = ({ text, percentage, className, transparent }: CChartSummeryProps) => {
+  let displayValue = '';
+  let arrowColor = '';
+  let arrowClass = '';
+
+  if (percentage) {
+    const bnPercentage = new BN(percentage);
+    const isPositive = bnPercentage.isGreaterThan(0);
+    displayValue = isPositive ? bnPercentage.toFixed() : bnPercentage.abs().toFixed();
+    arrowColor = isPositive ? '#027A48' : 'red';
+    arrowClass = isPositive ? '' : 'rotate-180';
+  }
+
+  return (
+    <div
+      className={classNames(
+        'w-fit rounded-md text-center',
+        className,
+        {
+          'bg-transparent': transparent,
+        },
+        { 'bg-[#F2F4F7]': !transparent },
+      )}
+    >
+      {text ? (
+        <div className="text-[#101828]">{text}</div>
+      ) : (
+        percentage && (
+          <div
```

Comment:
unacceptable code flow here. Fix it
The problem:

Reply:
not my code